### ISSUE TITLE
Triplet loss implementation

### ISSUE THREAD

**Issue by amirj** [explosion/thinc#500]
For a use case, I'm interested to use [triplet/hinge loss](https://en.wikipedia.org/wiki/Triplet_loss), looking at [the implementation of different losses in Thinc](https://github.com/explosion/thinc/blob/master/thinc/loss.py), I can't see something similar. Though there are various implementation in TensorFlow or PyTorch, it's not clear to me how can I leverage them to implement `get_grad` and `get_loss` functions.

**Comment by kadarakos** [comment]
Hey @amirj, 

Sorry for the late reply! Currently we do not support the Triplet loss, but you could try out [ `L2Distance` ](https://github.com/explosion/thinc/blob/master/thinc/loss.py#L295) and [`CosineDistance`](https://github.com/explosion/thinc/blob/master/thinc/loss.py#L325) losses. Both losses are applicable in the scenario where you are trying to map two spaces onto the same one. Does this help?
